geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/stats/DescriptiveStats.scala

```diff
@@ -0,0 +1,358 @@
+
+  override def +(that: DescriptiveStats): DescriptiveStats = {
+    val stats = new DescriptiveStats(attributes)
+    if (that.isEmpty) {
+      stats.copyFrom(this)
+    } else if (this.isEmpty) {
+      stats.copyFrom(that)
+    } else {
+      stats.copyFrom(this)
+      stats += that
+    }
+    stats
+  }
+
+  override def +=(that: DescriptiveStats): Unit = {
+    if (this == that)
+      return
+    if (that.isEmpty)
+      return
+    if (this.isEmpty) {
+      copyFrom(that)
+    } else {
+
+      val n1 = this._count
+      val n2 = that._count
+      val n1_squared = n1 * n1
+      val n2_squared = n2 * n2
+      val n_product = n1 * n2
+      val n = n1 + n2
+      val n_i = 1d / n
+
+      val delta = that._mean - this._mean
+
+      val A = delta * n_i
+      val A_squared = A * A
+
+      _m4n += that._m4n +
+        n_product * (n1_squared - n_product + n2_squared) * delta * A * A_squared +
+        (n1_squared * that._m2n + n2_squared * _m2n) * A_squared * 6d +
+        (n1 * that._m3n - n2 * _m3n) * A * 4d
+
+      _m3n += that._m3n +
+        n_product * (n1 - n2) * delta * A_squared +
+        ( n1 * that._m2n - n2 * _m2n) * A * 3d
+
+      _m2n += that._m2n +
+        n_product * delta * A
+
+      _c2 += (that._c2 + (delta |*| delta.T) * (n_product * n_i))
+
+      _mean += n2 * A
+
+      _sum += that._sum
+
+      _min.min(that._min)
+      _max.max(that._max)
+
+      _count += that._count
+    }
+  }
+
+  override def isEmpty: Boolean = _count < 1
+
+  override def isEquivalent(other: Stat): Boolean = other match {
+    case that: DescriptiveStats =>
+      attributes == that.attributes &&
+        _count  == that._count &&
+        _min.isIdentical(that._min, 1e-9) &&
+        _max.isIdentical(that._max, 1e-9) &&
+        _sum.isIdentical(that._sum, 1e-9) &&
+        _mean.isIdentical(that._mean, 1e-9) &&
+        _sum.isIdentical(that._sum, 1e-9) &&
+        _m2n.isIdentical(that._m2n, 1e-9) &&
+        _m3n.isIdentical(that._m3n, 1e-9) &&
+        _m4n.isIdentical(that._m4n, 1e-9) &&
+        _c2.isIdentical(that._c2, 1e-9)
+    case _ => false
+  }
+
+  override def toJson: String = {
+    val map = if (isEmpty) {
+      Map("count" -> 0)
+    } else {
+      ListMap("count" -> count,
+        "minimum" -> min,
+        "maximum" -> max,
+        "mean" -> mean,
+        "population_variance" -> pvar,
+        "population_standard_deviation" -> psdev,
+        "population_skewness" -> pskew,
+        "population_kurtosis" -> pkurt,
+        "population_excess_kurtosis" -> pekurt,
+        "sample_variance" -> svar,
+        "sample_standard_deviation" -> ssdev,
+        "sample_skewness" -> sskew,
+        "sample_kurtosis" -> skurt,
+        "sample_excess_kurtosis" -> sekurt,
+        "population_covariance" -> pcov,
+        "population_correlation" -> pcor,
+        "sample_covariance" -> scov,
+        "sample_correlation" -> scor)
+    }
+    Stat.JSON.toJson(map.asJava)
+  }
+}
+
+object SimpleMatrixUtils {
```

Comment:
We have made it this far without, but on the other hand, as we do more data science...  

We can refactor it later.